UnityProject/Assets/Scripts/Input System/KeyboardInputManager.cs

```diff
@@ -0,0 +1,231 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using static KeybindManager;
+
+public class KeyboardInputManager : MonoBehaviour
+{
+	public static KeyboardInputManager Instance;
+	private KeybindManager keybindManager => KeybindManager.Instance;
+
+	public enum KeyEventType
+	{
+		Down,
+		Up,
+		Hold
+	}
+	void Awake ()
+	{
+		if (Instance == null)
+		{
+			Instance = this;
+		}
+		else
+		{
+			Destroy(gameObject);
+		}
+	}
+
+	void Update()
+	{
+		CheckKeyboardInput();
+	}
+
+	void CheckKeyboardInput()
+	{
+		if (!UIManager.IsInputFocus && GameData.IsInGame && CustomNetworkManager.Instance.IsClientConnected())

```

Comment:
A note: `CustomNetworkManager.Instance.IsClientConnected()` is now obsolete. But that is not a problem here as we are waiting to see what the new unet holds before converting everything